### PR TITLE
Atom type sampling powerup

### PR DESCRIPTION
**Pull request by sblackburn86**
only update 1 atom during atom type update as an option &
greedy update for sample with at least 1 non-masked atom &
atom type update in corrector step

Note that we could add checks that the end of the generation to make sure all atoms transitioned to non-masked. it would be possible to have MASK atoms at the end if things go poorly.

### CODE REVIEW

File: src/diffusion_for_multi_scale_molecular_dynamics/generators/langevin_generator.py [hunk 1]
```diff
@@ -214,11 +228,97 @@ def atom_types_update(
...
+            ~all_masked.view(-1, 1),
+            unmask_this_sample,
+        )
+        do_greedy_sampling = torch.logical_and(do_greedy_sampling, atom_is_masked)
+        # replace the probability of getting a mask for those by 0 - so that stat cannot be sampled
+        one_step_transition_probs[:, :, -1] = torch.where(
+            do_greedy_sampling, 0, one_step_transition_probs[:, :, -1]
+        )
+
+        # replace u with a constant for samples with a non-MASK token present - this ensures a greedy sampling
```
Comment (rousseab) [inline src/diffusion_for_multi_scale_molecular_dynamics/generators/langevin_generator.py:318]: As we discussed, it might be useful to add a comment that this essentially means predicting MASK given our beta schedule...

**Review by rousseab — CHANGES_REQUESTED** [review]
The tests are failing. Please review.

File: src/diffusion_for_multi_scale_molecular_dynamics/generators/langevin_generator.py [hunk 2]
```diff
@@ -214,11 +228,97 @@ def atom_types_update(
...
+            atom_types_i: indices of atom types at time i. Dimension: [number_of_samples, number_of_atoms]
+            u: gumbel noise used for sampling. Dimension: [number_of_samples, number_of_atoms, num_classes]
+
+        Returns:
+            one_step_transition_probs: probabilities are updated so a MASK to non-MASK transition can happen
+            u: set to a constant for samples with at least 1 non-MASK atom
+        """
+        # check which samples have at least 1 non-MASK atom
+        all_masked = torch.all(
+            atom_types_i == self.num_classes - 1, dim=-1
```
[the remark this answers is not in the document]
Comment (rousseab) [reply src/diffusion_for_multi_scale_molecular_dynamics/generators/langevin_generator.py]: done

**Comment by rousseab** [comment]
I created a branch from commit 60ce99f and rebased on the `dev_atom_type` branch. I fixed the merge conflicts. I then merged this back into `dev_atom_type`. There are no more conflicts and all the tests pass.

I claim that this contains all the meaningful changes in branch. I believe the commits after 60ce99f do not contain meaningful changes, just conflict whackamole. I you agree, I propose that you close this branch.
